ts/modules/full-screen.src.ts

```diff
@@ -1,80 +1,237 @@
+                    ) : lang.exitFullscreen;
+        }
+    },
+    // Helping function - to open fullscreen, use toggleFullscreen() instead.
+    openFullscreen: function (): void {
+        const fullscreenController = this,
+            chart = fullscreenController.chart;
+
+        // Handle closeFullscreen() method when user clicks 'Escape' button.
+        chart.unbindFullscreenEvent = H.addEvent(
+            chart.container.ownerDocument, // chart's document
+            fullscreenController.browserProps.fullscreenChange,
+            function (): void {
+                // Handle lack of async of browser's fullScreenChange event.
+                if (chart.isFullscreen) {
+                    chart.isFullscreen = false;
+                    fullscreenController.closeFullscreen();
+                } else {
+                    chart.isFullscreen = true;
+                }
+            }
+        );
+
+        if (chart.container.parentNode instanceof Element) {
+            const promise = chart.container.parentNode[
+                fullscreenController.browserProps.requestFullscreen
+            ]();
+            if (promise) {
+                promise.catch(function (): void {
+                    alert( // eslint-disable-line no-alert
+                        'Full screen is not supported inside a frame'
+                    );
+                });
+            }
+            fullscreenController.setButtonText();
+        }
+
+        H.addEvent(chart, 'destroy', chart.unbindFullscreenEvent);
+    },
+    // Helping function - to close fullscreen, use toggleFullscreen() instead.
+    closeFullscreen: function (): void {
+        const fullscreenController = this,
+            chart = fullscreenController.chart;
+
+        // Don't fire exitFullscreen() when user exited using 'Escape' button.
+        if (
+            chart.isFullscreen &&
+            chart.container.ownerDocument instanceof Document
+        ) {
+            chart.container.ownerDocument[
+                fullscreenController.browserProps.exitFullscreen
+            ]();
+            this.setButtonText();
+        }
+
+        // Unbind event as it's necessary only before exiting from fullscreen.
+        if (chart.unbindFullscreenEvent) {
+            chart.unbindFullscreenEvent();
+        }
+
+        chart.isFullscreen = false;
+    },
+    /**
+     * Toggles displaying the chart in fullscreen mode.
+     * By default, when the exporting module is enabled, a context button with
+     * a drop down menu in the upper right corner accesses this function.
+     * Exporting module required.
+     *
+     * @since       next
+     *
+     * @sample      highcharts/members/chart-togglefullscreen/
+     *              Toggle fullscreen mode from a HTML button
+     *
+     * @function    Highcharts.Chart#toggleFullscreen
+     * @return      {void}
+     * @requires    modules/exporting
+     * @requires    modules/fullscreen
```

Comment:
It is `modules/full-screen`

Reply:
Actually, I think it is not needed at all because the `full-screen` module is a part of the `exporting` module - not a separate one.

Reply:
If you look into the masters, you will find a `full-screen` module, independent of the `exporting` module, as the `exporting` module is only required for the menu.

Reply:
@TorsteinHonsi `full-screen` is part of the `exporting` package, which looks to me like a bug, as there is nothing related between both except for the menu point. Can we remove it from there as a dependency?

Reply:
The conclusion is that we can not remove it from `exporting` package as it is now expected there.